src/pip/_internal/req/req_install.py

```diff
@@ -297,7 +334,8 @@ def _correct_build_location(self):
             return
         assert self.req is not None
         assert self._temp_build_dir.path
-        assert self._ideal_build_dir.path
+        # remove ignore after update to latest (0.650) mypy
+        assert self._ideal_build_dir is not None and self._ideal_build_dir.path  # type: ignore  # noqa: E501
```

Comment:
Please remove the # noqa here.

Reply:
@pradyunsg Could you explain what should be the right type of `self._ideal_build_dir`? It looks like string from all the usages, but this assert assumes it's `TempDirectory`.

Reply:
I think it should be `TempDirectory` but if it's `str` somewhere, that's a code smell identified then. :)